src/Chapter2.hs

```diff
@@ -727,7 +756,8 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate [] = []
+smartReplicate (x:xs) = (concat $ map (replicate x) [x]) ++ smartReplicate xs
```

Comment:
Great! 👍🏼 
If you want to write this function even more elegantly, you can use the standard function `concatMap` which is a combination of `concat` and `map` as the name suggests 🙂
After that, you could notice, that you would be able to eta-reduce on the last argument :)

Also, you are applying `map` to a single element list. This seems redundant, if you just apply function definitions step by step:

```haskell
concat $ map (replicate x) [x]
= concat [replicate x x]
= replicate x x
```

You can apply functions to the main list itself directly.